### PR TITLE
Migrate Graql Insert tests to BDD

### PR DESCRIPTION
**Pull request by alexjpwalker**
## What is the goal of this PR?

To migrate our Graql Insert tests to BDD.

This PR forms part of the work towards https://github.com/graknlabs/grakn/issues/5719 .

## What are the changes implemented in this PR?

Adds new BDD tests, many of which were migrated from GraqlInsertIT in Grakn Core.

### CODE REVIEW

File: behaviour/graql/language/insert.feature [hunk 1]
```diff
@@ -200,64 +295,1159 @@ Feature: Graql Insert Query
...
+    Then concept identifiers are
+      |     | check | value |
+      | PER | key   | ref:0 |
+      | EMP | key   | ref:1 |
+    Then uniquely identify answer concepts
+      | p   | r   |
+      | PER | EMP |
+
+
+  Scenario: when inserting a ternary relation that both owns an attribute and has an attribute as a roleplayer, both attributes are created
```
Comment (flyingsilverfin) [inline behaviour/graql/language/insert.feature:642]: ditch the `when`

Comment (flyingsilverfin) [reply behaviour/graql/language/insert.feature:642]: hmm actually maybe just a bit different:
"when inserting a relation that owns an attribute and has an attribute roleplayer, both attributes are created"

File: behaviour/graql/language/insert.feature [hunk 2]
```diff
@@ -200,64 +295,1159 @@ Feature: Graql Insert Query
...
+
+  Scenario: an additional role player can be inserted into a relation
+    Given graql insert
+      """
+      insert $p isa person, has ref 0; $r (employee: $p) isa employment, has ref 1;
+      """
+    Given the integrity is validated
+    When graql insert
+      """
+      match $r isa employment; insert $r (employer: $c) isa employment; $c isa company, has ref 2;
```
Comment (flyingsilverfin) [inline behaviour/graql/language/insert.feature:723]: split over two lines - easier to read if match/insert and separated

Comment (flyingsilverfin) [reply behaviour/graql/language/insert.feature:723]: (and elsewhere too)

File: behaviour/graql/language/insert.feature [hunk 3]
```diff
@@ -200,64 +295,1159 @@ Feature: Graql Insert Query
...
+    Then graql insert throws
+      """
+      insert
+      $r (employer: $p) isa employment, has ref 0;
+      $p isa person, has ref 1;
+      """
+    Then the integrity is validated
+
+
+  Scenario: inserting a roleplayer that can't play the role, yet is the parent of a type that can play the role, is still invalid and throws on commit
```
Comment (flyingsilverfin) [inline behaviour/graql/language/insert.feature:826]: shorten

File: behaviour/graql/language/insert.feature [hunk 4]
```diff
@@ -200,64 +295,1159 @@ Feature: Graql Insert Query
...
+    Then concept identifiers are
+      |     | check | value           |
+      | RES | key   | ref:0           |
+      | TEN | value | tenure-days:365 |
+    Then uniquely identify answer concepts
+      | a   |
+      | TEN |
+
+
+  Scenario: an attribute ownership currently inferred by a rule can be explicitly inserted
```
Comment (flyingsilverfin) [inline behaviour/graql/language/insert.feature:576]: should this be under `attribute insertion`?

File: behaviour/graql/language/insert.feature [hunk 5]
```diff
@@ -200,64 +295,1159 @@ Feature: Graql Insert Query
...
+      |     | check | value |
+      | MIC | key   | ref:1 |
+      | TAR | key   | ref:3 |
+    # Should only contain variables mentioned in the insert (so excludes `$z`)
+    Then uniquely identify answer concepts
+      | x   | y   |
+      | MIC | TAR |
+
+
+  Scenario: if match-insert matches nothing, then it inserts nothing
```
Comment (flyingsilverfin) [inline behaviour/graql/language/insert.feature:1199]: `then nothing is inserted` - lets leave out "it"

File: behaviour/graql/language/insert.feature [hunk 6]
```diff
@@ -200,64 +295,1159 @@ Feature: Graql Insert Query
...
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match $x isa person, has name "Derek"; get;
+      """
+    Then answer size is: 0
+
+
+  Scenario: if any insert in a transaction fails with a semantic error, none of the inserts are performed
```
Comment (flyingsilverfin) [inline behaviour/graql/language/insert.feature:1393]: this makes me think we may want to decide to catch a "specific" type of exception, what do you think?

File: behaviour/graql/language/insert.feature [hunk 7]
```diff
@@ -26,172 +26,267 @@ Feature: Graql Insert Query
...
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match $x has name "John"; get;
+      """
+    Then answer size is: 0
+
+
+  Scenario: when inserting a new thing that owns an existing attribute, and that attribute has no previous owner, that thing becomes its first owner
```
[the remark this answers is not in the document]
Comment (alexjpwalker) [reply behaviour/graql/language/insert.feature:259]: `given an attribute with no owners, inserting a thing that owns it results in it having an owner`

File: behaviour/graql/language/insert.feature [hunk 8]
```diff
@@ -26,172 +26,267 @@ Feature: Graql Insert Query
...
+      """
+    Then concept identifiers are
+      |      | check | value |
+      | KYLE | key   | ref:0 |
+    Then uniquely identify answer concepts
+      | x    |
+      | KYLE |
+
 
+  Scenario: after inserting two things that own the same attribute, the things become linked, in that they are both owners of that attribute
```
[the remark this answers is not in the document]
Comment (alexjpwalker) [reply behaviour/graql/language/insert.feature:282]: `after inserting two things that own the same attribute, the attribute has two owners`

File: behaviour/graql/language/insert.feature [hunk 9]
```diff
@@ -200,64 +295,1159 @@ Feature: Graql Insert Query
...
+      $x <val1> isa <attr>, has ref 0;
+      $y <val2> isa <attr>, has ref 1;
+      $p isa person, has name "Imogen", has ref 2, has <attr> <val1>, has <attr> <val2>;
+      """
+    When the integrity is validated
+    When get answers of graql query
+      """
+      match $p isa person, has <attr> $x; get $x;
+      """
+    Then concept identifiers are
```
[the remark this answers is not in the document]
Comment (alexjpwalker) [reply behaviour/graql/language/insert.feature:365]: Unfortunately it doesn't work for `datetime` attributes, as it tries to compare `"2020-01-01"` to `"2020-01-01T00:00"` and finds that they don't match. Specifying that full-length date string in the`concept identifiers are` table doesn't work either, as the colon breaks the table formatting.

File: behaviour/graql/language/insert.feature (hunk 4 above)
Comment (alexjpwalker) [reply behaviour/graql/language/insert.feature:576]: Could be either really. I think the key thing is that we're inserting an ownership, not just an attribute.

File: behaviour/graql/language/insert.feature (hunk 1 above)
Comment (alexjpwalker) [reply behaviour/graql/language/insert.feature:642]: Done!

File: behaviour/graql/language/insert.feature (hunk 2 above)
Comment (alexjpwalker) [reply behaviour/graql/language/insert.feature:723]: Done

File: behaviour/graql/language/insert.feature (hunk 3 above)
Comment (alexjpwalker) [reply behaviour/graql/language/insert.feature:826]: `inserting a roleplayer that is only the parent of a valid type is invalid and throws on commit`

File: behaviour/graql/language/insert.feature (hunk 5 above)
Comment (alexjpwalker) [reply behaviour/graql/language/insert.feature:1298]: done

File: behaviour/graql/language/insert.feature (hunk 6 above)
Comment (alexjpwalker) [reply behaviour/graql/language/insert.feature:1492]: Probably yes - but that could be applied to literally hundreds of our tests and I feel like it should be deferred for now.

Comment (alexjpwalker) [reply behaviour/graql/language/insert.feature:1500]: Raised https://github.com/graknlabs/verification/issues/60.

**Pull request merged**